src/Common/Extensions/StringExtensions.cs

```diff
@@ -74,5 +75,29 @@ public static IList<string> Chunkify(this string str, int maxLength)
 
             return result;
         }
+
+        /// <summary>
+        /// Hash a string using SHA-256
+        /// </summary>
+        /// <param name="str">The string to hash.</param>
+        /// <returns>The SHA-256 hash of the given string.</returns>
+        public static string Hash(this string str)
+        {
+            if (string.IsNullOrEmpty(str))
+            {
+                return str;
+            }
+
+            var sha = SHA256.Create();
```

Comment:
We might need to call `Dispose` on this instance https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha256?view=netcore-2.1